subsys/bluetooth/services/hids_c.c

```diff
@@ -0,0 +1,851 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+#include <kernel.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/uuid.h>
+#include <bluetooth/gatt.h>
+
+#include <bluetooth/services/hids_c.h>
+
+#define LOG_LEVEL CONFIG_BT_GATT_HIDS_C_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(hids_c);
+
+/* Suspend value for Control Point  */
+#define BT_GATT_HIDS_C_CP_SUSPEND      0x00
+/* Exit suspend value for Control Point */
+#define BT_GATT_HIDS_C_CP_EXIT_SUSPEND 0x01
+
+/* Memory slab used for reports */
+K_MEM_SLAB_DEFINE(bt_gatt_hids_c_reports_mem,
+		  sizeof(struct bt_gatt_hids_c_rep_info),
+		  CONFIG_BT_GATT_HIDS_REPORTS_MAX,
+		  4);
+
+
+/**
+ * @brief Allocate memory for report structure
+ *
+ * Function allocates memory for report structure.
+ * The structure would be cleared.
+ *
+ * @return The pointer to the structure or NULL if no memory is available.
+ *
+ * @sa rep_new
+ */
+static struct bt_gatt_hids_c_rep_info *rep_alloc(void)
+{
+	struct bt_gatt_hids_c_rep_info *rep;
+	if (k_mem_slab_alloc(&bt_gatt_hids_c_reports_mem,
+			     (void**)&rep, K_NO_WAIT)) {
```

Comment:
explicit cast to void should not be needed or is it in this case...

Reply:
I agree with you, but the compiler has different opinion:

```
hids_c.c:43:9: warning: passing argument 2 of ‘k_mem_slab_alloc’ from incompatible pointer type [-Wincompatible-pointer-types]
         &rep, K_NO_WAIT)) {
```